### PR TITLE
[MERGE] various: support batch creation in marketing apps

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
SPECIFICATIONS

By default, ORM supports creation of records in batch (one SQL INSERT by
record but batching other operations). When a create override on a given
model is specified as api.model or api.model_create_single this ability
is disabled and records are created one by one. This leads to performance
losses.

In this merge we update create to support batch creation. This allows to
gain performances linked to batch code improvements. Note that some of the
changes done in this merge will not have much impact because some records
are always created one by one. Converting them
  * encourages new overrides to support multi record creation;
  * provides the basis for a potential future API change to enforce batch
    creation support;

LINKS

Task ID-2330149
COM PR odoo/odoo#61246
ENT PR odoo/enterprise#14561

### CODE REVIEW

**Review by kos94ok-3D — COMMENTED** [review]
fetch a field value in one query

File: addons/website_slides_forum/models/slide_channel.py [hunk 1]
```diff
@@ -25,12 +25,11 @@ def action_redirect_to_forum(self):
...
-    @api.model
-    def create(self, vals):
-        channel = super(Channel, self.with_context(mail_create_nosubscribe=True)).create(vals)
-        if channel.forum_id:
-            channel.forum_id.privacy = False
-        return channel
+    @api.model_create_multi
+    def create(self, vals_list):
+        channels = super(Channel, self.with_context(mail_create_nosubscribe=True)).create(vals_list)
+        channels.forum_id.privacy = False
```
Comment (kos94ok-3D) [inline addons/website_slides_forum/models/slide_channel.py:31]: missing check that `forum_id` is set

File: addons/product/models/res_company.py [hunk 2]
```diff
@@ -7,24 +7,28 @@
...
+            pricelist = ProductPricelist.search([
+                ('currency_id', '=', new_company.currency_id.id),
+                ('company_id', '=', False)
+            ], limit=1)
+            if not pricelist:
+                params = {'currency': new_company.currency_id.name}
+                pricelist = ProductPricelist.create({
+                    'name': _("Default %(currency)s pricelist") %  params,
+                    'currency_id': new_company.currency_id.id,
+                })
```
Comment (kos94ok-3D) [inline addons/product/models/res_company.py:24]: <pre>
currency_companies = self.env['res.company'].read_group([
    ('id', 'in', companies.ids),
], ['currency_id', 'company_ids:array_agg(id)'], ['currency_id'])
for vals in currency_companies:
    currency_id = vals['currency_id'][0]
    pricelist = ProductPricelist.search([
        ('currency_id', '=', currency_id),
        ('company_id', '=', False),
    ], limit=1)
    if not pricelist:
        currency_name = self.env['res.currency'].browse(currency_id)
        for new_company in self.browse(vals['company_ids']):
            params = {'currency': currency_name}
            pricelist = ProductPricelist.create({
                'name': _("Default %(currency)s pricelist") %  params,
                'currency_id': currency_id,
            })
</pre>

**Review by kos94ok-3D — COMMENTED** [review]
Use read_group

File: addons/website_slides_forum/models/slide_channel.py (hunk 1 above)
Comment (Feyensv) [reply addons/website_slides_forum/models/slide_channel.py:31]: Calling write on an empty recordset is supported (and has no impact)

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

File: addons/website_forum/models/forum.py [hunk 3]
```diff
@@ -180,11 +180,14 @@ def _compute_count_flagged_posts(self):
...
-        res = super(Forum, self.with_context(mail_create_nolog=True, mail_create_nosubscribe=True)).create(values)
-        res._set_default_faq()  # will trigger a write and call update_website_count
-        return res
+    @api.model_create_multi
+    def create(self, vals_list):
+        forums = super(
+            Forum,
+            self.with_context(mail_create_nolog=True, mail_create_nosubscribe=True)
+        ).create(vals_list)
+        forums._set_default_faq()  # will trigger a write and call update_website_count
```
Comment (JKE-be) [inline addons/website_forum/models/forum.py:189]: Another free change, never tested ! :clap: @Feyensv 

1 commit, for 1 "free" diff, for 1 new bug.

Fixed by https://github.com/odoo/odoo/pull/76463

Comment (tde-banana-odoo) [reply addons/website_forum/models/forum.py:189]: It was tested. You may refer to the task and see it followed the classic process. Issue was not spotted but you cannot say "never tested".

**Pull request closed without merging**